apps/chat/agent/tools.py

```diff
@@ -124,24 +124,21 @@ def _move_datetime_to_new_weekday_and_time(date: datetime, new_weekday: int, new
     day_diff = new_weekday - current_weekday
     return date.replace(hour=new_hour, minute=new_minute, second=0) + timedelta(days=day_diff)
 
-
-def create_periodic_task(experiment_session: ExperimentSession, message: str, **kwargs):
-    task_kwargs = json.dumps(
-        {
-            "chat_ids": [experiment_session.participant.identifier],
-            "message": message,
-            "is_bot_instruction": False,
-            "experiment_public_id": str(experiment_session.experiment.public_id),
-        }
-    )
-    PeriodicTask.objects.create(
-        name=f"reminder-{experiment_session.id}-{uuid.uuid4()}",
-        task=BOT_MESSAGE_FOR_USER_TASK,
-        kwargs=task_kwargs,
-        **kwargs,
+def create_schedule_message(experiment_session: ExperimentSession, message: str, trigger_date: datetime, **kwargs):
+    from apps.utils.factories.events import EventActionFactory
+    additional_params = {
+        "name": f"reminder-{experiment_session.id}-{uuid.uuid4()}",
+        'prompt_text': message,
+    }
+    params = {**kwargs, **additional_params}
+    ScheduledMessage.objects.create(
+        participant=experiment_session.participant,
+        team=experiment_session.team,
+        action=EventActionFactory(params=params, action_type=EventActionType.SCHEDULETRIGGER),
```

Comment:
I don't think we should be using factories outside of test code? Maybe one can in principle, but I would be very careful, since there may be auto generated fields in there.